### DISCUSSION TITLE
Parameterized  UID and PRODID

### DISCUSSION THREAD

**Discussion by vinayak** [add2cal/add-to-calendar-button#189] [Q&A] [answered]
Can we customize UID and PRODID by passing as parameter?

**Comment by jekuer** [comment]
Hi @vinayak ,

to be completely honest, no.
The PRODID is meant to hold information about the generating entity, which basically is this tool/script. Therefore, this part needs to hold the copyright and source information of the generator.
The UID has a slightly different purpose, but used for similar things. Therefore, it also should hold the generator information.
Since users usually never see this "source code" information, I do not see a problem here.
The current mode helps for debugging as well as being transparent. Theoretically, one could even argue changing this would be not in line with the underlying license.

Best,
Jens

**Reply by vinayak** [reply]
I can understand for PRODID, but UID we use for modifying the event information, like reschedule, cancel and updating the content in combination with SEQUENCE.

**Reply by jekuer** [reply]
Can you describe this scheme a little bit better?
You can set the sequence already. Maybe we should make the status configurable too?

**Reply by vinayak** [reply]
In our calendar invite we use combination of UID and sequence to update the event. suppose we create an event, and later if the event gets reschedule or cancelled the previous invite should move or removed from the calendar.

**Reply by jekuer** [reply]
yes, but how exactly.
Or let me guess: the underlying requirement is that you want to be able to define the unique identifier, so any updated ics gets matched against the original event, right?

In addition, you would still need the option to set the event status, which is currently not possible with this script.

This would make sense to me and is definitely worth implementing :)

**Reply by jekuer** [reply]
#190

**Reply by vinayak** [reply]
yes you are right I need a reference for updating the ics.  and also update the status. I do agree you can have PRODID reserved for generating application. But need UID and status to be customized.

**Reply by jekuer** [reply]
Check then. Makes sense. Will implement.

**Reply by vinayak** [reply]
Thanks @jekuer

**Comment by jekuer** [comment, marked as answer]
Updated in new version 1.16
